Ninjathulhu/Ninjathulhu.Game/Entity.cs

```diff
@@ -68,6 +68,30 @@ public class EntityProperties
         public Dictionary<Type, ComponentProperties> FromPrefab = new Dictionary<Type, ComponentProperties>();
 
         public Dictionary<Type, ComponentProperties> FromSpawn = new Dictionary<Type, ComponentProperties>();
+
+        private static object Get(Dictionary<Type, ComponentProperties> properties, Type componentType, string propertyName)
+        {
+            if (properties.ContainsKey(componentType) &&
+                properties[componentType].Values.ContainsKey(propertyName))
+            {
+                return properties[componentType].Values[propertyName];
+            }
+
+            return null;
+        }
+
+        public object Get<TComponent>(string propertyName, object defaultValue = null)
+        {
+            object result;
+
+            result = Get(FromSpawn, typeof(TComponent), propertyName);
+            if (result != null) { return result; }
+
+            result = Get(FromPrefab, typeof(TComponent), propertyName);
+            if (result != null) { return result; }
+
+            return defaultValue;
```

Comment:
I see what you're saying.  This is kind of content-facing code, so I'm applying a different sort of logic.  Components should generally have sensible defaults for their properties so that prefabs can be lazily authored.